src/lib/input/_input-theme.scss

```diff
@@ -64,7 +64,7 @@
     }
   }
 
-  .mat-input-container.ng-invalid.ng-touched:not(.mat-focused) {
+  .mat-input-invalid {
```

Comment:
Mostly because I wanted to re-use the same logic that determines whether the messages are shown. These classes get forwarded "manually" from the input element anyway so there shouldn't be much of a performance difference.